Plugins/Sycl/src/Seeding/CreateSeedsForGroupSycl.cpp

```diff
@@ -440,54 +587,106 @@ void createSeedsForGroupSycl(
 
-        // nd_range with maximum block size for triplet search and filter
-        // global and local range is given
-        cl::sycl::nd_range<1> tripletSearchNDRange{
-            cl::sycl::range<1>(numWorkItems(numCombinations, maxWorkGroupSize)),
-            cl::sycl::range<1>(maxWorkGroupSize)};
+        cl::sycl::nd_range<1> tripletFilterNDRange =
+            calculate1DimNDRange(numTripletFilterThreads, maxWorkGroupSize);
 
-        auto tripletEvent = q->submit([&](cl::sycl::handler& h) {
+        sycl::buffer<uint32_t> countTripletsBuf(deviceCountTriplets.data(),
+                                                edgesBottom);
+
+        auto tripletKernel = q->submit([&](cl::sycl::handler& h) {
           h.depends_on({linB, linT});
+          auto countTripletsAcc =
+              sycl::ONEAPI::atomic_accessor<uint32_t, 1,
+                                            sycl::ONEAPI::memory_order::relaxed,
+                                            sycl::ONEAPI::memory_scope::device>(
+                  countTripletsBuf, h);
+          auto numTopDupletsAcc = numTopDupletsBuf.get_access<
+              sycl::access::mode::read, sycl::access::target::global_buffer>(h);
           h.parallel_for<triplet_search_kernel>(
               tripletSearchNDRange, [=](cl::sycl::nd_item<1> item) {
                 const uint32_t idx = item.get_global_linear_id();
-                if (idx < numCombinations) {
-                  // Retrieve the index of the corresponding middle space point
-                  // by binomial search
+                if (idx < numTripletSearchThreads) {
+                  // Retrieve the index of the corresponding middle
+                  // space point by binary search
                   auto L = firstMiddle;
                   auto R = lastMiddle;
                   auto mid = L;
                   while (L < R - 1) {
                     mid = (L + R) / 2;
-                    if (idx + tripletSearchOffset[0] < deviceSumComb[mid]) {
+                    // To be able to search in deviceSumComb, we need
+                    // to use an offset (sumCombUptoFirstMiddle).
+                    if (idx + sumCombUptoFirstMiddle < deviceSumComb[mid]) {
                       R = mid;
                     } else {
                       L = mid;
                     }
                   }
                   mid = L;
 
-                  detail::TripletData T = {MIN, MIN};
-                  deviceCurvImpact[idx] = T;
-                  const auto numT = deviceNumTopDuplets[mid];
+                  const auto numT = numTopDupletsAcc[mid];
+                  const auto threadIdxForMiddleSP =
+                      (idx - deviceSumComb[mid] + sumCombUptoFirstMiddle);
+
+                  // NOTES ON THREAD MAPPING TO SPACE POINTS
+                  /*
+                    We need to map bottom and top SP indices to this
+                    thread. This is done in the following way: We
```

Comment:
it sounds like you want to map multiple SP to one thread, but it's only one triplet per thread, right?

Reply:
Right, I clarified it a little. :)